test/libsolidity/semanticTests/abiEncoderV1/abi_encode_calldata_slice.sol

```diff
@@ -59,10 +59,10 @@ contract C {
 // EVMVersion: >homestead
 // ----
 // test_bytes() ->
-// gas irOptimized: 360863
-// gas legacy: 411269
-// gas legacyOptimized: 317754
+// gas irOptimized: 321332
+// gas legacy: 319111
+// gas legacyOptimized: 269317
```

Comment:
Looks like gains are sometimes non-negligible even in situations that aren't completely unrealistic. `test_bytes()` does 36 external calls (to the same contract, but still) and you'd think that would dominate the cost but apparently the cost of loop counter increments is still significant in comparison.

Reply:
It's not without reason that people actually do the horrible ``unchecked``-block increment construction. But that also means, that in actual real-world situations, we likely won't see gas-improvements, but rather code-improvements (since the workaround is no longer necessary)